controlplane/src/core/composition/schemaLint.ts

```diff
@@ -0,0 +1,231 @@
+        break;
+      }
+      case 'SHOULD_NOT_HAVE_TYPE_SUFFIX': {
+        rulesConfig[ruleName] = [rule.severity, { ObjectTypeDefinition: { forbiddenSuffixes: ['Type', 'type'] } }];
+        break;
+      }
+      case 'SHOULD_NOT_HAVE_INPUT_PREFIX': {
+        rulesConfig[ruleName] = [
+          rule.severity,
+          { InputObjectTypeDefinition: { forbiddenPrefixes: ['Input', 'input'] } },
+        ];
+        break;
+      }
+      case 'SHOULD_HAVE_INPUT_SUFFIX': {
+        rulesConfig[ruleName] = [rule.severity, { InputObjectTypeDefinition: { requiredSuffixes: ['Input'] } }];
+        break;
+      }
+      case 'SHOULD_NOT_HAVE_ENUM_PREFIX': {
+        rulesConfig[ruleName] = [
+          rule.severity,
+          { EnumTypeDefinition: { forbiddenPrefixes: ['Enum', 'enum', 'ENUM'] } },
+        ];
+        break;
+      }
+      case 'SHOULD_NOT_HAVE_ENUM_SUFFIX': {
+        rulesConfig[ruleName] = [rule.severity, { EnumTypeDefinition: { forbiddenSuffixes: ['Enum', 'enum'] } }];
+        break;
+      }
+      case 'SHOULD_NOT_HAVE_INTERFACE_PREFIX': {
+        rulesConfig[ruleName] = [
+          rule.severity,
+          { InterfaceTypeDefinition: { forbiddenPrefixes: ['Interface', 'interface'] } },
+        ];
+        break;
+      }
+      case 'SHOULD_NOT_HAVE_INTERFACE_SUFFIX': {
+        rulesConfig[ruleName] = [
+          rule.severity,
+          { InterfaceTypeDefinition: { forbiddenSuffixes: ['Interface', 'interface'] } },
+        ];
+        break;
+      }
+      case 'ENUM_VALUES_SHOULD_BE_UPPER_CASE': {
+        rulesConfig[ruleName] = [rule.severity, { EnumValueDefinition: { style: 'UPPER_CASE' } }];
+        break;
+      }
+      case 'DISALLOW_CASE_INSENSITIVE_ENUM_VALUES': {
+        rulesConfig[ruleName] = [rule.severity];
+        break;
+      }
+      case 'NO_TYPENAME_PREFIX_IN_TYPE_FIELDS': {
+        rulesConfig[ruleName] = [rule.severity];
+        break;
+      }
+      case 'REQUIRE_DEPRECATION_REASON': {
+        rulesConfig[ruleName] = [rule.severity];
+        break;
+      }
+      case 'REQUIRE_DEPRECATION_DATE': {
+        rulesConfig[ruleName] = [rule.severity];
+        break;
+      }
+      case 'ORDER_FIELDS': {
+        rulesConfig[ruleName] = [
+          rule.severity,
+          { fields: ['ObjectTypeDefinition', 'InterfaceTypeDefinition', 'InputObjectTypeDefinition'] },
+        ];
+        break;
+      }
+      case 'ORDER_ENUM_VALUES': {
+        rulesConfig[ruleName] = [rule.severity, { values: ['EnumTypeDefinition'] }];
+        break;
+      }
+      case 'ORDER_DEFINITIONS': {
+        rulesConfig[ruleName] = [rule.severity, { definitions: true }];
+        break;
+      }
+      case 'ALL_TYPES_REQUIRE_DESCRIPTION': {
+        rulesConfig[ruleName] = [rule.severity, { types: true }];
+        break;
+      }
+      default: {
+        throw new Error(`Rule ${rule} doesnt exist`);
+      }
+    }
+  }
+  return rulesConfig;
+};
+
+const directiveDefinitions = `
```

Comment:
I think we should reference to the composition library. The library needs to maintain this to compose and validate correctly. This can drift.

Reply:
Done